### DISCUSSION TITLE
GoogleMapsToOSMAndGPX utility - convert a custom google map you've created directly into OSMAnd compatible GPX files.

### DISCUSSION THREAD

**Discussion by tmusolf** [osmandapp/OsmAnd#19899] [Show and tell]
It's a python utility that works directly from a google map (using it's map ID) to export the map's KML data and directly convert it into a folder of OSMAnd style GPX files. Both tracks and waypoints and translated.   Descriptions, icon symbol, icon color, track color, track width are all translated. Each track is put in it's own GPX file and all waypoints are put in a single GPX file.

**Comment by Sonwon1** [comment]
I would find this useful!

**Comment by tmusolf** [comment]
I've updated this utility to convert almost all GMap icons.  It also optionally processes layers, creating separate folders of GPX files for each layer. 

There are now 2 windows executables, one is a GUI wrapper and the other is the utility itself so you don't need to have python installed.

**Reply by Sonwon1** [reply]
Thank you, will give this a test.
